calico_containers/integrations/kubernetes/tests/kube_plugin_test.py

```diff
@@ -0,0 +1,378 @@
+
+    def setUp(self):
+        # Mock out sh when importing calico_kubernetes so it doesn't fail when
+        # trying to find the calicoctl binary (which may not exist)
+        with patch('calico_containers.integrations.'
+                      'kubernetes.calico_kubernetes.sh') as m_sh:
+            self.plugin = calico_kubernetes.NetworkPlugin()
+
+    def test_create(self):
+        with patch.object(self.plugin, '_configure_interface') as m_interface, \
+                patch.object(self.plugin, '_configure_profile') as m_prof:
+            # Set up mock objects
+            m_interface.return_value = 'endpt_id'
+
+            # Set up args
+            args = ['script name','setup','','pod-name','dockerId']
+
+            # Call method under test
+            self.plugin.create(args)
+
+            # Assert
+            self.assertEqual('pod_name', self.plugin.pod_name)
+            self.assertEqual('dockerId', self.plugin.docker_id)
+            m_interface.assert_called_once_with()
+            m_prof.assert_called_once_with('endpt_id')
+
+    def test_create_fail(self):
+        with patch.object(self.plugin, '_configure_interface') as m_interface, \
+                patch.object(self.plugin, '_configure_profile') as m_prof, \
+                patch('calico_containers.integrations.'
+                      'kubernetes.calico_kubernetes.sys.exit') as m_sys:
+            # Set up mock objects
+            m_interface.side_effect = CalledProcessError(1,'','')
+
+            # Set up args
+            args = ['script name','setup','','podname','dockerId']
+
+            # Call method under test
+            self.plugin.create(args)
+
+            # Assert
+            m_sys.assert_called_once_with(1)
+
+    def test_delete(self):
+        with patch.object(self.plugin, 'calicoctl') as m_calicoctl:
+            # Set up args
+            args = ['script name','teardown', '','pod-name','dockerId']
+
+            # Call method under test
+            self.plugin.delete(args)
+
+            # Assert
+            self.assertEqual(self.plugin.pod_name, 'pod_name')
+            self.assertEqual(self.plugin.docker_id, 'dockerId')
+            m_calicoctl.assert_any_call('container', 'remove', 'dockerId')
+            m_calicoctl.assert_any_call('profile', 'remove', 'pod_name')
+
+    def test_configure_interface(self):
+        with patch.object(self.plugin, '_read_docker_ip') as m_read_docker, \
+                patch.object(self.plugin, '_delete_docker_interface') as m_delete_docker, \
+                patch.object(calico_kubernetes, 'container_add') as m_add_container, \
+                patch.object(calico_kubernetes, 'generate_cali_interface_name') as m_cali, \
+                patch.object(self.plugin, '_get_node_ip') as m_node_ip, \
+                patch.object(calico_kubernetes, 'check_call') as m_check_call:
+            # Set up mock objects
+            m_read_docker.return_value = 'docker_ip'
```

Comment:
Its a good idea to use realistic values for your mocks.  In this case, we expect "_read_docker_ip" to return an IP address formatted as a string, rather than just any old string.